components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -15,28 +21,91 @@ import mozilla.components.support.base.log.logger.Logger
     private var experimentsLoaded: Boolean = false
-    private val evaluator = ExperimentEvaluator(valuesProvider)
-    private val logger = Logger(LOG_TAG)
+    private var evaluator: ExperimentEvaluator = ExperimentEvaluator(ValuesProvider())
+    private lateinit var logger: Logger
+
+    private var source: ExperimentSource? = null
+    private var storage: ExperimentStorage? = null
+
+    /**
+     * Initialize the experiments library.
+     *
+     * This should only be initialized once by the application.
+     *
+     * @param applicationContext [Context] to access application features, such
+     * as shared preferences.
+     */
+    fun initialize(
+        applicationContext: Context
+    ) {
+        experimentsResult = ExperimentsSnapshot(listOf(), null)
+        experimentsLoaded = false
+        logger = Logger(LOG_TAG)
+
+        if (source == null) {
+            source = KintoExperimentSource(
+                EXPERIMENTS_BASE_URL,
+                EXPERIMENTS_BUCKET_NAME,
+                EXPERIMENTS_COLLECTION_NAME,
+                HttpURLConnectionClient()
+            )
+        }
+        if (storage == null) {
+            storage = FlatFileExperimentStorage(
+                // TODO: is this the right directory?
+                File(applicationContext.filesDir,
+                EXPERIMENTS_JSON_FILENAME)
+            )
+        }
+    }
+
+    /**
+     * Override the experiment source for testing.
+     *
+     * @param aSource the experiment source to use
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun overrideExperimentSource(aSource: ExperimentSource) {
+        source = aSource
+    }
+
+    /**
+     * Override the experiment storage for testing.
+     *
+     * @param aStorage the experiment storage to use
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun overrideExperimentStorage(aStorage: ExperimentStorage) {
+        storage = aStorage
+    }
+
+    /**
+     * Override the values provider for testing.
+     *
+     * @param valuesProvider the values provider to use
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun overrideValuesProvider(valuesProvider: ValuesProvider) {
+        evaluator = ExperimentEvaluator(valuesProvider)
+    }
 
     /**
      * Provides the list of experiments (active or not)
      */
-    val experiments: List<Experiment>
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal val experiments: List<Experiment>
         get() = experimentsResult.experiments.toList()
 
     /**
      * Loads experiments from local storage
      */
     @Synchronized
-    fun loadExperiments() {
-        experimentsResult = storage.retrieve()
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun loadExperiments() {
+        experimentsResult = storage!!.retrieve()
```

Comment:
The `!!` here throws an exception if `storage` is `null` (I understand this should never happen outside of tests, unless the API is called before `initialize`). If we want the API to not ever throw exception (if that's a concern), we might consider refactoring this to something like:

```
storage?.let {
  experimentsResult = it.retrieve()
  experimentsLoaded = true
} ?: logger.error("No storage specified")
```

Or something like that.